client/reader/controller.js

```diff
@@ -306,7 +316,10 @@ module.exports = {
 					onClose: function() {
 						page.back( context.lastRoute || '/' );
 					},
-					onClosed: removeFullPostDialog
+					onClosed: removeFullPostDialog,
+					onPostNotFound: function() {
+						renderPostNotFound( i18n.translate( 'Feed post not found' ) );
```

Comment:
these can both be just "Post not found". There's no diff to user if it's a feed post or a blog post.